framework/CodeInterfaces/SCALE/ScaleInterface.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April 04, 2018
+
+@author: alfoa
+
+comments: Interface for Scale Simulation (current Origen and Triton)
+"""
+from __future__ import division, print_function, unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default',DeprecationWarning)
+
+import os
+import copy
+import shutil
+from utils import utils
+import xml.etree.ElementTree as ET
+
+from GenericCodeInterface import GenericParser
+from CodeInterfaceBaseClass import CodeInterfaceBase
+from tritonAndOrigenData import origenAndTritonData
+
+class Scale(CodeInterfaceBase):
+  """
+    Scale Interface. It currently supports Triton and Origen sequences only.
+  """
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    CodeInterfaceBase.__init__(self)
+    self.sequence = []  # this contains the sequence that needs to be run. For example, ['triton'] or ['origen'] or ['triton','origen']
+    self.timeUOM  = 's' # uom of time (i.e. s, h, m, d, y )
+    self.outputRoot = {'triton':None,'origen':None} # the root of the output sequences
+
+  def _readMoreXML(self,xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class and initialize
+      some members based on inputs. This can be overloaded in specialize code interface in order to
+      read specific flags
+      @ In, xmlNode, xml.etree.ElementTree.Element, Xml element node
+      @ Out, None.
+    """
+    CodeInterfaceBase._readMoreXML(self,xmlNode)
+    sequence = xmlNode.find("sequence")
+    if sequence is None:
```

Comment:
should we error out if sequence is none?

Reply:
No, because by default it is "triton"